internal/pkg/cli/dialog/create_workspace.go

```diff
@@ -0,0 +1,94 @@
+package dialog
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/keboola/go-client/pkg/sandbox"
+	"github.com/keboola/keboola-as-code/internal/pkg/cli/options"
+	"github.com/keboola/keboola-as-code/internal/pkg/cli/prompt"
+	workspace "github.com/keboola/keboola-as-code/pkg/lib/operation/project/remote/workspace"
+)
+
+type createWorkspaceDeps interface {
+	Options() *options.Options
+}
+
+func (p *Dialogs) AskCreateWorkspace(d createWorkspaceDeps) (workspace.CreateOptions, error) {
+	opts := workspace.CreateOptions{}
+
+	name, err := p.askWorkspaceName(d)
+	if err != nil {
+		return opts, err
+	}
+	opts.Name = name
+
+	typ, err := p.askWorkspaceType(d)
+	if err != nil {
+		return opts, err
+	}
+	opts.Type = typ
+
+	if sandbox.SupportsSizes(typ) {
+		size, err := p.askWorkspaceSize(d)
+		if err != nil {
+			return opts, err
+		}
+		opts.Size = size
+	}
+
+	return opts, nil
+}
+
+func (p *Dialogs) askWorkspaceName(d createWorkspaceDeps) (string, error) {
+	if d.Options().IsSet("name") {
+		return d.Options().GetString("name"), nil
+	} else {
+		name, ok := p.Ask(&prompt.Question{
+			Label:     "Enter a name for the new workspace",
+			Validator: prompt.ValueRequired,
+		})
+		if !ok || len(name) == 0 {
+			return "", fmt.Errorf("missing name, please specify it")
+		}
+		return name, nil
+	}
+}
+
+func (p *Dialogs) askWorkspaceType(d createWorkspaceDeps) (string, error) {
+	if d.Options().IsSet("type") {
+		typ := d.Options().GetString("type")
+		if _, found := sandbox.TypesMap[typ]; !found {
```

Comment:
Kedze je to `map[...]bool`, tak sa to da zapisat aj ako:
```
if sandbox.TypesMap[typ] {
```

**Je to jedno**, len poznamka, ze `map` vracia `zero value` (tj. v tomto pripade `false`), ak kluc neexistuje, ... 

Z map sa da citat tak, ze tam mas aj `found`, alebo bez (to je default)

```
v := m[key]
v, found := m[key]
```